Rework Workflow Documentation

maximiliankolb opened the pull request:
First draft for Issue #6398

mdellweg commented on docs/workflows/sync.rst:
```diff
@@ -1,91 +1,161 @@
+-------------------------------
 
-``$ http POST $BASE_ADDR/pulp/api/v3/remotes/deb/apt/ name='bar' url='http://some.url/somewhere/'``
+Use the remote object to kick off a synchronize task by specifying the repository to sync with.
```
It's the other way around. The remote is the parameter to the sync endpoint on the repository.

mdellweg commented on docs/workflows/upload.rst:
```diff
@@ -1,94 +1,344 @@
+This requires you to already have a :ref:`repository created <create_a_repository>`.
 
-    $ http POST $BASE_ADDR/pulp/api/v3/repositories/deb/apt/ name=foo
+If you don't have any repositories yet, create one:
```
Can we move all of this to a `create_a_repository` section and just state: We assume you have a repository named `vim`.
This section should really focus on the package upload.

mdellweg commented on docs/workflows/upload.rst:
```diff
@@ -1,94 +1,344 @@
+
+.. note::
+
+  A common pool location will be generated if you do not specify a ``relative_path``.
```
Read that before...
Would it be enough to say something like: "Instead of specifying an existing artifact, you can upload a file with the following command. The rest is the same."

quba42 commented:
I am in favour of creating a sub task of https://pulp.plan.io/issues/6398 for this PR.
Something along the lines of "Update the workflow documentation", since that accurately reflects the scope of this PR.

maximiliankolb commented:
Thanks for your initial review @mdellweg. Would you please have a look at it again as I incorporated all your suggestions and fixed the format of the PR.
Any feedback is welcome.

I would very much like to move forward with the documentation and possibly add and extend it in the future.

CC @quba42

quba42 commented:
I will have a look at it (as soon as my pulplift boxes are up and I have had a chance to look at some of the other pulp_deb PRs).

quba42 commented on CHANGES/6398.doc:
```diff
@@ -0,0 +1 @@
+Reworked workflow and added flow charts
```
I created a new ticket that more accurately reflects the scope of this change: https://pulp.plan.io/issues/7148
(I tried to create it as a sub task of 6389, but that did not work!?)
I would also suggest changing the text to something like:
```
Reworked the workflow documentation and added flow charts.
```
I find this clearer even if it is already implied by the `.doc` extension.
